src/legacy/shared/GoogleMapsWithMarkers/GoogleMapsWithMarkers.tsx

```diff
@@ -208,3 +220,31 @@ function getCenterCoordinates(listings: ListingShort[]) {
     lng: lng / (length || 1),
   };
 }
+
+function getCenterOfDirections(directions: google.maps.DirectionsResult) {
+  const path = directions.routes[0].overview_path;
+  const distances = path.map((point, index) => index === 0 ? 0 : Math.sqrt(
+    Math.pow(path[index - 1].lat() - point.lat(), 2) +
+    Math.pow(path[index - 1].lng() - point.lng(), 2)
+  ));
+  const total = distances.reduce((a, b) => a + b, 0);
+  let start = 0;
+  let travelled = 0;
+  while (travelled + distances[start] < total / 2) {
+    travelled += distances[start];
+    start += 1;
+  }
+  const remaining = total - travelled;
+  const endWeight = remaining / distances[start];
+  const startWeight = 1 - endWeight;
+  return {
+    lat: startWeight * path[start].lat() + endWeight * path[start + 1].lat(),
+    lng: startWeight * path[start].lng() + endWeight * path[start + 1].lng()
+  };
```

Comment:
This function is currently pretty darn broken; unfortunately, there's not a "center of this route" property on the directions result.